beziers.scad

```diff
@@ -711,18 +710,19 @@ function bezier_path_length(path, N=3, max_deflect=0.001) =
 //   ];
 //   trace_path(bez, size=1, N=3, showpts=true);
 //   trace_path(bezier_path(bez, N=3), size=3);
-function bezier_path(bezier, splinesteps=16, N=3) =
+function bezier_path(bezier, splinesteps=16, N=3, endpoint=true) =
     assert(is_path(bezier))
     assert(is_int(N))
     assert(is_int(splinesteps))
-    assert(len(bezier)%N == 1, str("A degree ",N," bezier path shound have a multiple of ",N," points in it, plus 1."))
+    assert(len(bezier)%N == 1, str("A degree ",N," bezier path should have a multiple of ",N," points in it, plus 1."))
     let(
-        segs = (len(bezier)-1)/N
-    ) deduplicate([
-        for (seg = [0:1:segs-1], i = [0:1:splinesteps-1])
-            bezier_path_point(bezier, seg, i/splinesteps, N=N),
-        bezier_path_point(bezier, segs-1, 1, N=N)
-    ]);
+        segs = (len(bezier)-1) / N,
+        step = 1 / splinesteps
+    ) [
+        for (seg = [0:1:segs-1])
+            each bezier_points(select(bezier, seg*N, (seg+1)*N), [0:step:1-step/2]),
```

Comment:
Assume you're avoiding lerpn here for efficiency?  The closing point if endpoint=true should just be last(bezier) instead of a call to bezier_path_point.  Guaranteed exact and faster.

Reply:
Good point.